src/plugins/wizard/public/application/components/data_tab/field_selector.tsx

```diff
@@ -39,7 +32,23 @@ const META_FIELDS: string[] = [
 
 export const FieldSelector = () => {
   const indexFields = useTypedSelector((state) => state.dataSource.visualizableFields);
-  const fields = indexFields?.reduce<IFieldCategories>(
+  const [filteredFields, setFilteredFields] = useState(indexFields);
+  const [fieldSearchValue, setFieldSearchValue] = useState('');
+
+  useEffect(() => {
+    const filteredSubset = indexFields.filter((field) => {
+      if (!field.displayName.includes(fieldSearchValue)) {
+        return false;
+      }
+
+      return true;
+    });
```

Comment:
Nit: this can be just a one liner of you are only checking to see if the display name includes the search term. The waterfall approach is more useful if there are more conditions you want to filter on.

```
const filteredSubset = indexFields.filter((field) => field.displayName.includes(fieldSearchValue))
```

Reply:
Done.